profiles/views.py

```diff
@@ -66,29 +67,37 @@ def post(self, request, user_id=None):
         username = request.user
         detail = None
 
-        if permissions.is_in_group(username, 'StudentGroup'):
-            user = get_object_or_404(User, pk=username.id)
-            user_form = UserForm(request.POST, instance=user)
-            detail = get_object_or_404(StudentDetail, user=user)
-            detail_form = StudentForm(request.POST, instance=detail)
+        if username == request.user:
+            request.POST = request.POST.copy()
 
-        elif permissions.is_in_group(username, 'FacultyGroup'):
-            user = get_object_or_404(User, pk=username.id)
-            user_form = UserForm(request.POST, instance=user)
-            detail = get_object_or_404(FacultyDetail, user=user)
-            detail_form = FacultyForm(request.POST, instance=detail)
+            # A simple hack to stop someone from editing username 
+            request.POST['username'] = username
```

Comment:
Please remove this and remove the username field from the UserForm. That way, it will work.